src/Libraries/Storage/UploadedFile.php

```diff
@@ -377,17 +389,35 @@ protected function moveUploadedFile(string $filePath): bool
     }
 
     /**
-     * Whitelist the file extension
+     * Validates upload against allowed mime types => extensions map
      * @param string $extension
+     * @param string $mimeType
      * @return bool
      */
-    protected function whitelisted(string $extension): bool
+    protected function allowed(string $extension, string $mimeType): bool
     {
-        if (!preg_match('/(' . implode('|', $this->blacklistedExtensions) . ')$/i', $extension)) {
-            return true;
-        }
+        $extension = strtolower($extension);
+        $mimeType = strtolower($mimeType);
 
-        return false;
+        return isset($this->allowedMimeTypes[$mimeType]) &&
+            in_array($extension, $this->allowedMimeTypes[$mimeType], true);
+    }
+
+    /**
+     * Loads allowed mime types from config (shared/config/uploads.php) if present.
+	 * @return void
+     */
+    protected function loadAllowedMimeTypesFromConfig(): void
+    {
+		if (!config()->has('uploads')) {
+			config()->import(new Setup('config', 'uploads'));
+		}
+
+        $allowedMimeTypesMap = config()->get('uploads.allowed_mime_types');
+
+        if (is_array($allowedMimeTypesMap)) {
```

Comment:
This will silently ignore the config, better to throw exception if the config is not in correct structure